smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/ServiceGenerator.kt

```diff
@@ -68,6 +73,42 @@ class ServiceGenerator(
                 outputShapeName
             )
         }
+
+        /**
+         * Helper method for generating in-line documentation for operation
+         */
+        private fun renderOperationDoc(model: Model, service: ServiceShape, op: OperationShape, writer: SwiftWriter) {
+            writer.writeShapeDocs(op)
+            writer.writeAvailableAttribute(model, op)
+
+            writer.writeSingleLineDocs { write("") }
+            writer.writeDocs("\\- Parameter ${op.inputShape.name} : ${retrieveMemberShapeDoc(op.inputShape, model)}")
+
+            writer.writeSingleLineDocs { write("") }
+            writer.writeDocs("\\- Returns: \\`${op.outputShape.name}\\` : ${retrieveMemberShapeDoc(op.outputShape, model)}")
+
+            if (op.getErrors(service).size > 0) {
+                writer.writeSingleLineDocs { write("") }
+                writer.writeSingleLineDocs { write("- Throws: `${op.toUpperCamelCase() + "Error"}` : Place-holder wrapper object for possible exceptions listed below.") }
+                writer.writeSingleLineDocs { write("") }
+                writer.writeSingleLineDocs { write("__Possible Exceptions:__") }
+            }
+
+            for (error in op.getErrors(service)) {
+                writer.writeDocs("\\- \\`${error.name}\\` : ${retrieveMemberShapeDoc(error.toShapeId(), model)}")
+            }
```

Comment:
This is probably a matter of preference but I would prefer

```
val errorDocs = op.getErrors(service).joinToString("\n").forEach { error ->
    "\\- \\`${error.name}\\` : ${retrieveMemberShapeDoc(error.toShapeId(), model)}"
}
writer.writeDocs(errorDocs)
```

The above code utilizes Kotlin syntax as well as separates the concerns of string formation and doc writing

Reply:
Given that it seems to be a matter of preference as you suggested, I think I'll keep the for loop format as I don't think changing this is worth the possible unexpected behavior / quirks of SwiftWriter I came across last week. I've changed all other syntax comments to reflect your comments tho :D